### DISCUSSION TITLE
run st2 with docker-compose, failed to connect to mongo

### DISCUSSION THREAD

**Discussion by gudan803** [StackStorm/st2#5941] [Operations]
rabbitmq/mongo/redis are running OK, but st2 apps  are always restarting.
![image](https://user-images.githubusercontent.com/28525794/224669190-7fc1224c-13dc-45b2-baa9-c1304dcf6ae3.png)
![image](https://user-images.githubusercontent.com/28525794/224669295-9917b956-090f-4d78-b84c-9e7a0f75a2ab.png)

**Comment by gudan803** [comment]
database st2 failed to initialize
![image](https://user-images.githubusercontent.com/28525794/225192101-cae50592-a030-4463-a029-ce95f76edaae.png)

**Comment by arm4b** [comment]
Do you have any custom changes in st2.conf or docker-compose? or is it a default config

Also check the docker networking and if `mongo` hos is available from the st2 containers.
Worth checking the MongoDB logs too if the DB started with any issues.
